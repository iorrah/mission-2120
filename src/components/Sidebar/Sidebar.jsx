import React from "react";

const Sidebar = function() {
  return (
    <aside className="sidebar">
      <div className="sidebar__skeleton"></div>

      <div className="sidebar__content">
        <img src="img/icon-2.png" className="sidebar__icon" alt="Dashboard" />
        <img src="img/icon-3.png" className="sidebar__icon" alt="Dashboard" />
        <img src="img/icon-9.png" className="sidebar__icon" alt="Dashboard" />
        <img src="img/icon-4.png" className="sidebar__icon" alt="Dashboard" />
        <img src="img/icon-5.png" className="sidebar__icon" alt="Dashboard" />
        <img src="img/icon-6.png" className="sidebar__icon" alt="Dashboard" />
        <img src="img/icon-7.png" className="sidebar__icon" alt="Dashboard" />
      </div>
    </aside>
  );
};

export default Sidebar;
