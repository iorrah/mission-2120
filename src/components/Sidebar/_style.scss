.sidebar {
  flex-grow: 1;
}

.sidebar__skeleton {
  background-color: $white;
  position: absolute;
  width: 3.4rem;
  height: 100%;
  top: 0;
  z-index: 0;
}

.sidebar__content {
  width: 3.4rem;
  overflow: hidden;
  height: 100%;
  padding: 1.4rem 0 0;
  position: relative;
  z-index: 1;
}

.sidebar__icon {
  width: 1.7rem;
  display: block;
  margin: 0 auto 1.9rem;
  cursor: pointer;
}
