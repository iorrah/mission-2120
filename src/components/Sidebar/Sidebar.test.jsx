import React from "react";
import { render } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import Sidebar from "./Sidebar";

describe("Sidebar", () => {
  test("should shallow correctly", () => {
    expect(shallow(<Sidebar />)).toMatchSnapshot();
  });

  test("should mount correctly", () => {
    expect(mount(<Sidebar />)).toMatchSnapshot();
  });

  test("should render correctly", () => {
    expect(render(<Sidebar />)).toMatchSnapshot();
  });

  it("renders the `.Sidebar` div", () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper.find(".sidebar").exists()).toBeTruthy();
  });

  it("renders the white background `.sidebar__skeleton`", () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper.find(".sidebar__skeleton").exists()).toBeTruthy();
  });

  it("renders the `.sidebar__content` wrapper", () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper.find(".sidebar__content").exists()).toBeTruthy();
  });

  it("renders 7 `.sidebar__icon`", () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper.find(".sidebar__icon").length).toBe(7);
  });
});
