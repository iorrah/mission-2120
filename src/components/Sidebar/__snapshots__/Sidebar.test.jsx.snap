// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should mount correctly 1`] = `ReactWrapper {}`;

exports[`Sidebar should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="sidebar"
      >
        <div
          class="sidebar__skeleton"
        />
        <div
          class="sidebar__content"
        >
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-2.png"
          />
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-3.png"
          />
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-9.png"
          />
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-4.png"
          />
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-5.png"
          />
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-6.png"
          />
          <img
            alt="Dashboard"
            class="sidebar__icon"
            src="img/icon-7.png"
          />
        </div>
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside
      class="sidebar"
    >
      <div
        class="sidebar__skeleton"
      />
      <div
        class="sidebar__content"
      >
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-2.png"
        />
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-3.png"
        />
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-9.png"
        />
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-4.png"
        />
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-5.png"
        />
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-6.png"
        />
        <img
          alt="Dashboard"
          class="sidebar__icon"
          src="img/icon-7.png"
        />
      </div>
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Sidebar should shallow correctly 1`] = `ShallowWrapper {}`;
